src/deps_compat.jl

```diff
@@ -1,21 +1,24 @@
 """
-    Aqua.test_deps_compat(package)
+    Aqua.test_deps_compat(package; [ignore])
 
 Test that `Project.toml` of `package` list all `compat` for `deps`.
 
 # Arguments
 - `packages`: a top-level `Module`, a `Base.PkgId`, or a collection of
   them.
+
+# Keyword Arguments
+- `ignore::Vector{Symbol}`: names of dependent packages to be ignored.
 """
 test_deps_compat
-function test_deps_compat(packages)
-    @testset "$(result.label)" for result in analyze_deps_compat(packages)
+function test_deps_compat(packages; kwargs...)
+    @testset "$(result.label)" for result in analyze_deps_compat(packages; kwargs...)
         @debug result.label result
         @test result ⊜ true
     end
 end
 
-analyze_deps_compat(packages) = [_analyze_deps_compat_1(pkg) for pkg in aspkgids(packages)]
+analyze_deps_compat(packages; ignore) = [_analyze_deps_compat_1(pkg) for pkg in aspkgids(packages) if !(Symbol(p.name) in ignore)]
```

Comment:
I don't think this can pass tests if you don't provide a default value. Also, maybe specify the type?
```suggestion
analyze_deps_compat(packages; ignore::Vector{Symbol} = Symbol[]) = [_analyze_deps_compat_1(pkg) for pkg in aspkgids(packages) if !(Symbol(p.name) in ignore)]
```

Reply:
(Or maybe `AbstractVector{Symbol}`, to match `_analyze_stale_deps_1`)

Reply:
Done! I've added a default value, and I've specified the type as `AbstractVector{Symbol}`.